File: day01.py
import math

## PART ONE
res = 0
for input in open('input.txt', encoding='utf-8'):
    res += (math.floor(int(input) / 3) - 2)

print(res)

## PART TWO
res2 = 0
for input in open('input.txt', encoding='utf-8'):
    toAdd = 0
    counting = True
    base = (math.floor(int(input) / 3) - 2)
    toAdd += base
    while counting:
        count = (math.floor(int(base) / 3) - 2)
        if count < 0:
            counting = False
        else:
            toAdd += count
            base = count
    res2+=toAdd

print(res2)

File: day01.js
var fs = require('fs');
var read = fs.readFileSync("input.txt");
var data = read.toString().split("\n").map(Number);
data.pop();

function calculate(el){
    return Math.floor(el/3)-2;
}

// PART ONE 
console.log(data.map(el => calculate(el)).reduce( (a,b) => a+b));

// PART TWO 
console.log(data.map(sumItUp).reduce( (a,b) => a+b))

function sumItUp(el){
    let toAdd = 0;
    counting = true;
    base = calculate(el);
    toAdd += base;

    while(counting){
        count = calculate(base);
        count < 0 ? counting = false : (toAdd+=count, base=count);
    }
    return toAdd
}